### ISSUE TITLE
`PlatformException.message` should be treated like `PlatformException.stacktrace`

### ISSUE THREAD

**Issue by ueman** [getsentry/sentry-dart#1708]
### Problem Statement

In theory, only [`PlatformException.stacktrace`](https://api.flutter.dev/flutter/services/PlatformException/stacktrace.html) should contain a stacktrace. However, in practice I see a lot of plugins misusing the [`PlatformException.message`](https://api.flutter.dev/flutter/services/PlatformException/message.html) field for stacktraces.

### Solution Brainstorm

Therefore, it would be nice if the `PlatformException.message` field could be treated like the `PlatformException.stacktrace` field.

### Are you willing to submit a PR?

Yes

**Comment by buenaflor** [comment]
Do you have an example of how it's misused? Not sure if I understand correctly if `message` doesn't contain a stacktrace

**Comment by ueman** [comment]
Sure.

Here's an example where a native stacktrace is passed to Flutter in the `PlatformException.message` field:

<img width="1127" alt="Screenshot 2023-11-02 at 10 40 05" src="https://github.com/getsentry/sentry-dart/assets/1270149/f65efcc5-6b75-4090-a631-36270799fb4c">

As you can see, it's not really well formatted and Sentry could do better.

Here's an example of how it looks like when the native stacktrace is passed via the `PlatformException.stacktrace` field and how I want it to look like for the example above

<img width="1140" alt="Screenshot 2023-11-02 at 10 41 56" src="https://github.com/getsentry/sentry-dart/assets/1270149/006167c9-6247-446f-a5b9-9fdeb6891526">

It should be a straight forward change in the following file https://github.com/getsentry/sentry-dart/blob/955541a50762f04c301c83eb25aef0077742c29a/flutter/lib/src/event_processor/android_platform_exception_event_processor.dart

Basically the new logic should be in case `PlatformException.stacktrace` is null, try to parse `PlatformException.message` instead.

**Comment by buenaflor** [comment]
I see, makes sense if it is parsed as a fallback 👍
